core/src/orderbook.rs

```diff
@@ -22,6 +23,15 @@ pub trait StableXOrderBookReading: Send + Sync {
         batch_id_to_solve: u32,
     ) -> BoxFuture<'a, Result<(AccountState, Vec<Order>)>>;
 
+    /// Returns the state of the open orderbook at the specified block number.
+    ///
+    /// The open orderbook contains orders and balances that are valid for the
+    /// current batch at the given block.
+    fn get_auction_data_for_block<'a>(
```

Comment:
For consistency we could rename the other method to `get_auction_data_for_batch`